File: Java_tests/question1.java
package Java_tests;

import java.util.Arrays;
import java.util.Random;


public class question1 {

    public static void main(String[] args) {
        // Goal: Count how many ODD numbers are in a randomly generated array
        // of integers.
        //
        // Steps:
        //   1. Loop through the array and check each number one at a time. If
        //      the number is odd, add 1 to the oddCount variable.
        int[] numbers = numberGenerator(10, 1, 100);
        int oddCount = 0;

        // your code here ----




        // Print the result
        System.out.println("Question 1 - Odd Count");
        System.out.println("    Numbers: " + Arrays.toString(numbers));
        System.out.println("    Odd Count: " + oddCount);
    }
    

    private static Boolean isOdd(int number) {
        // Helper method that checks if a number is odd
        boolean result = number % 2 != 0;
        return result;
    }

    private static int[] numberGenerator(int size, int min, int max) {
        // Helper method that generates an array of random integers
        Random random = new Random();
        int[] numbers = new int[size];
        for (int i = 0; i < size; i++) {
            numbers[i] = random.nextInt(max - min + 1) + min;
        }
        return numbers;
    }
}

File: Python_questions/question1.py
import random


def question1():
    """"
    Goal: Count how many ODD numbers are in a randomly generated array
    of integers.

    Steps:
        1. Loop through the array and check each number one at a time.
        2. If the number is odd, add 1 to the odd_count variable.

    """

    numbers = number_generator(10, 1, 100)
    odd_count = 0

    # Your code here ↓↓↓↓




    # ↑↑↑↑ Your code goes above

    print("Question 1 - Odd Count")
    print(f"    Numbers:   {', '.join(numbers)}")
    print(f"    Odd Count: {odd_count}")


def is_odd(number):
    """Helper function that checks if a number is odd"""
    result = number % 2 != 0
    return result


def number_generator(size, min_val, max_val):
    """Helper function that generates a list of random integers"""
    numbers = []
    for i in range(size):
        numbers.append(random.randint(min_val, max_val))
    return numbers


if __name__ == "__main__":
    question1()
